viewing_party/party.py

```diff
@@ -0,0 +1,147 @@
+    user_data; returns updated user_data'''
+    user_data['watched'].append(movie) 
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    '''takes in a user_data dict with key 'watchlist' and a value which is a 
+    list of dicts repping movies the user wants to watch and a movie to add to 
+    user_data; returns updated user_data'''
+    user_data['watchlist'].append(movie) 
+    return user_data
+
+def watch_movie(user_data, title):
+    '''takes in a user_data dict with two keys--watchlist and watched--that 
+    each correspond to a list of movies, and a title; if title in watchlist,
+    it's removed from watchlist + added to watched; returns user_data'''
+    for movie in user_data['watchlist']:
+        if title == movie['title']:
+            user_data['watchlist'].remove(movie)
+            add_to_watched(user_data, movie)
+    return user_data
+
+def get_watched_avg_rating(user_data):
+    '''takes in a user_data dict with key 'watched' and values of list of 
+    watched movies; returns average rating of all movies'''
+    rating_sum = 0
+    num_of_movies = len(user_data['watched']) 
+    for movie in user_data['watched']:
+        rating_sum += movie['rating']
+    return 0.0 if num_of_movies == 0 else rating_sum/num_of_movies
+
+def get_most_watched_genre(user_data):
+    '''takes in a user_data dict with key 'watched' and values of list of 
+    watched movies; returns which genre occurs most frequently in list'''
+    if user_data['watched'] == []: return None 
+    
+    genre_dict = {}
+    for movie in user_data['watched']:
+        genre = movie['genre']
+        if genre in genre_dict:
+            genre_dict[genre] += 1
+        else:
+            genre_dict[genre] = 1
+
+    return max(genre_dict, key=genre_dict.get)  
+
+# extra helper method that's not asked for but was made to DRY up code 
+def find_user_and_friend_movies(user_data):
+    '''takes in a user_data dict with keys 'watched' (list of watched movies)
+    and friends (list of friends and movies they've watched); returns two 
+    lists--one that reps movies only user has watched and one that reps 
+    movies only friends have watched'''
+    movies_user_watched = [movie['title'] for movie in user_data['watched']]
+
+    movies_friends_watched = []
+    for friend in user_data['friends']:
+        for movie in friend['watched']:
+            movies_friends_watched.append(movie['title'])
+    
+    return movies_user_watched, movies_friends_watched
+
+def get_unique_watched(user_data):
+    '''takes in a user_data dict with keys 'watched' (list of watched movies)
+    and friends (list of friends and movies they've watched); returns list of 
+    dicts that rep list of movies that user has watched but friends haven't''' 
+    user_movies, friend_movies = find_user_and_friend_movies(user_data)
+    user_unique_movies = set(user_movies) - set(friend_movies)
+    return [{ 'title' : movie } for movie in user_unique_movies]
```

Comment:
verrry interesting! now it may not be very efficient since you are rebuilding the entire dictionary again, but I love the idea. You tried a new data structure we taught you and made it work! clever!